lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm

```diff
@@ -716,8 +716,247 @@ sub add_read_counts{
+
+          #Store unique combinations of patient, event type, data type and gene (e.g., PNC4  snv  wgs  KRAS)
+          $data_type_sum{$patient}{$event_type}{$data_type}{$gene_name} = 1;
+
+          #Add patient lists specific to this event type
+          if (defined($genes{$gene_name}{$event_type})){
+            my $patients = $genes{$gene_name}{$event_type}{patient_list};
+            $patients->{$patient} = 1;
+          }else{
+            my %patients;
+            $patients{$patient} = 1;
+            $genes{$gene_name}{$event_type}{patient_list} = \%patients;
+          }
+
+          if (defined($interactions{$interaction}{$event_type})){
+            my $patients = $interactions{$interaction}{$event_type}{patient_list};
+            $patients->{$patient} = 1;
+          }else{
+            my %patients;
+            $patients{$patient} = 1;
+            $interactions{$interaction}{$event_type}{patient_list} = \%patients;
+          }
+
+          #Create or update the grand list of patients with ANY events hitting this gene
+          if (defined($genes{$gene_name}{grand_list})){
+            my $patients = $genes{$gene_name}{grand_list};
+            $patients->{$patient} = 1;
+          }else{
+            my %patients;
+            $patients{$patient} = 1;
+            $genes{$gene_name}{grand_list} = \%patients;
+          }
+
+          if (defined($interactions{$interaction}{grand_list})){
+            my $patients = $interactions{$interaction}{grand_list};
+            $patients->{$patient} = 1;
+          }else{
+            my %patients;
+            $patients{$patient} = 1;
+            $interactions{$interaction}{grand_list} = \%patients;
+          }
+        }
+        close(IN);
+      }
+    }
+  }
+  $result{genes} = \%genes;
+  $result{interactions} = \%interactions;
+  $result{data_type_sum} = \%data_type_sum;
+  return(\%result);
+}
 
+sub get_somatic_subject_common_name {
+  my $self = shift;
+  my $b = shift;
+  my $subject_common_name = "null";
+  if($b->model->wgs_model) {
+    $subject_common_name = $b->model->wgs_model->last_succeeded_build->subject->common_name;
+  } elsif($b->model->exome_model) {
+    my $subject_common_name = $b->model->exome_model->last_succeeded_build->subject->common_name;
+  }
+  return $subject_common_name;
+}
 
+#Get input files to be parsed
+sub getFiles{
+  my $self = shift;
+  my %args = @_;
+  my $builds = $args{'-builds'};
+  my %files;
+
+  $self->status_message("\n\nGet annotation files and drug-gene interaction files from these builds");
+  foreach my $b (@$builds){
+    my $build_directory = $b->data_directory;
+    my $subject_common_name = $b->subject->common_name;
+    my $subject_name = $self->get_somatic_subject_common_name($b);
+    $subject_name =~ s/[\s-]/_/g;
+    my $build_id = $b->id;
+
+    #If the subject name is not defined, die
+    unless ($subject_name){
+      die $self->error_message("\n\nCould not determine subject name for build: $build_id\n\n");
```

Comment:
I recommend removing the extra `\n`s from these messages.  They separate the message from the `ERROR:` that precedes them, which makes the logs more confusing to follow.